### PR TITLE
chore(cli): Make sentry only send errors on panic

### PR DESCRIPTION
**Pull request by yevgenypats**
Go doesn't really have a great way of distinguishing exceptions and errors. Also we don't want to go into the rabbit hole of classifying every error as user error or internal error as sometimes it's not really possible: Let's say we had an error in our marshaling logic where usually this error means a user error in some cases it can be also an internal one.

We will have to rely on user reports for the most, apart from panics which are clearly internal errors and we would like to send them to sentry.

### CODE REVIEW

File: cli/main.go [hunk 1]
```diff
@@ -11,9 +13,18 @@ import (
...
+			sentry.Flush(sentryFlushTimeout)
+			fmt.Println(originalMessage)
+			os.Exit(1)
+		}
+	}()
+
 	if err := cmd.NewCmdRoot().Execute(); err != nil {
-		sentry.CaptureMessage(err.Error())
-		sentry.Flush(sentryFlushTimeout)
 		os.Exit(1)
```
Comment (disq) [inline cli/main.go:28]: This is fine, would just add a // lint: nolint here

File: cli/main.go [hunk 2]
```diff
@@ -10,10 +12,19 @@ import (
...
+			panic(err)
+		}
+	}()
+
 	if err := cmd.NewCmdRoot().Execute(); err != nil {
-		sentry.CaptureMessage(err.Error())
-		sentry.Flush(sentryFlushTimeout)
 		os.Exit(1)
 	}
 	sentry.Flush(sentryFlushTimeout)
```
Comment (disq) [inline cli/main.go:30]: No need for this line any more, and the const can also be removed.

File: cli/main.go [hunk 3]
```diff
@@ -10,10 +12,19 @@ import (
 
 const sentryFlushTimeout = 5 * time.Second
 
+//nolint:gocritic
```
Comment (disq) [inline cli/main.go:15]: Wouldn't add nolint to the whole fn but to the os.Exit line?

File: cli/main.go (hunk 2 above)
Comment (yevgenypats) [reply cli/main.go:29]: It is actually needed becayse call to `CaptureMessage` are not sent synchronously by default , but I changed the transport to sync now so we can avoid it as we call sentry only in one place

File: cli/main.go (hunk 3 above)
Comment (yevgenypats) [reply cli/main.go]: done

File: cli/main.go (hunk 2 above)
Comment (disq) [reply cli/main.go:30]: We still don't this line because if it panics and we're sending a sentry report in the defer, this line won't run anyway.

Comment (yevgenypats) [reply cli/main.go:30]: Oh I see. you commented on another line. sorry.

**Review by disq — APPROVED** [review]

**Pull request merged**